controllers/preflightvalidationocp_reconciler.go

```diff
@@ -0,0 +1,271 @@
+	return &PreflightValidationOCPReconciler{
+		client:             client,
+		filter:             filter,
+		registry:           registry,
+		registryAuthGetter: registryAuthGetter,
+		kernelOsDtkMapping: kernelOsDtkMapping,
+		statusUpdater:      statusUpdater,
+		scheme:             scheme,
+	}
+}
+
+func (r *PreflightValidationOCPReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("preflightvalidationocp").
+		For(&kmmv1beta1.PreflightValidationOCP{}, builder.WithPredicates(filter.PreflightOCPReconcilerUpdatePredicate())).
+		Watches(
+			&source.Kind{Type: &kmmv1beta1.PreflightValidation{}},
+			handler.EnqueueRequestsFromMapFunc(r.filter.EnqueueAllPreflightValidationsOCP),
+		).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: 1,
+		}).
+		Complete(r)
+}
+
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidations,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidations/status,verbs=get;list;watch
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidations/finalizers,verbs=update
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidationsocp,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidationsocp/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=kmm.sigs.k8s.io,resources=preflightvalidationsocp/finalizers,verbs=update
+
+// Reconcile Reconiliation entry point
+func (r *PreflightValidationOCPReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Start PreflightValidationOCP Reconciliation")
+
+	pvo := kmmv1beta1.PreflightValidationOCP{}
+	err := r.client.Get(ctx, req.NamespacedName, &pvo)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Reconciliation object not found; not reconciling")
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "preflight validation ocp reconcile failed to find object")
+		return ctrl.Result{}, err
+	}
+
+	err = r.runPreflightValidationOCP(ctx, &pvo)
+	if err != nil {
+		log.Error(err, "runPreflightValidationOCP failed")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *PreflightValidationOCPReconciler) runPreflightValidationOCP(ctx context.Context, pvo *kmmv1beta1.PreflightValidationOCP) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	// get compatible PreflightValidation
+	nsn := types.NamespacedName{Name: pvo.Name, Namespace: pvo.Namespace}
+	pv := &kmmv1beta1.PreflightValidation{}
+	err := r.client.Get(ctx, nsn, pv)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Compatible PreflightValidation not found, creating")
+			pv, err = r.preparePreflightValidation(ctx, pvo)
+			if err != nil {
+				return fmt.Errorf("failed to prepare the data for compatible PreflightValidation: %v", err)
+			}
+			err = r.client.Create(ctx, pv)
```

Comment:
Isn't [`CreateOrPatch`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#CreateOrPatch) and option here? We use it extensively in other controllers.

Reply:
Currently i do not see a scenario where user would like to update a Preflight or PreflightOCP. If user wants another kernel version/release image verifed, then he should create a new Preflight/ Preflight OCP CR

Reply:
That sounds a bit counter-intuitive to me, unless we make the field immutable (which seems only doable with a webhook). If you insist, we will need to make sure that this ends up in the docs.